synapseadmin/responses.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2023 Tulir Asokan
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package synapseadmin
+
+import (
+	"maunium.net/go/mautrix"
+	"maunium.net/go/mautrix/id"
+)
+
+// RoomInfos represents the structure of each room
+type RoomInfos struct {
+    RoomID             id.RoomID	`json:"room_id"`
+    Name               string		`json:"name"`
+    CanonicalAlias     string		`json:"canonical_alias"`
+    JoinedMembers      int    		`json:"joined_members"`
+    JoinedLocalMembers int    		`json:"joined_local_members"`
+    Version            string 		`json:"version"`
+    Creator            string 		`json:"creator"`
+    Encryption         string 		`json:"encryption"`
```

Comment:
Some of these should have proper types:

CanonicalAlias: `id.RoomAlias`
Creator: `id.UserID`
Encryption: `id.Algorithm`